File: myschedule-web/src/main/resources/myschedule/script/examples/simpleJobs.rb
# Groovy Examples

# Using myschedule.quartz.extra.SchedulerTemplate API
# ===================================================
require 'java'
java_import Java::MyscheduleQuartzExtraJob::LoggerJob
java_import Java::MyscheduleQuartzExtraJob::ScriptingJob
java_import Java::MyscheduleQuartzExtra::SchedulerTemplate
java_import Java::JavaUtil::Date
java_import Java::JavaLang::System

# Schedule hourly job
scheduler.schedule_simple_job("hourlyJob", -1, 60 * 60 * 1000, LoggerJob.java_class)

# Schedule minutely job
scheduler.schedule_simple_job("minutelyJob", -1, 60 * 1000, LoggerJob.java_class)

# Schedule secondly job
scheduler.schedule_simple_job("secondlyJob", -1, 1000, LoggerJob.java_class)

# Schedule secondly job that repeat total of 3 times.
scheduler.schedule_simple_job("secondlyJobRepeat3", 3, 1000, LoggerJob.java_class)

# Schedule onetime job that run immediately
scheduler.schedule_simple_job("onetimeJob", 1, 0, LoggerJob.java_class)

# Schedule hourly job with job data and start time of 20s delay.
scheduler.schedule_simple_job("hourlyJobWithStartTimeDelay", -1, 60 * 60 * 1000, ScriptingJob.java_class, 
		SchedulerTemplate.mk_map(
			'ScriptEngineName', 'Groovy', 
			'ScriptText', '''
				logger.info("I am a script job...")
				sleep(700L)
				logger.info("I am done.")
			'''
		), 
		Date.new(System.current_time_millis() + 20 * 1000))

# Schedule one job with multiple triggers
require 'java'
java_import Java::OrgQuartz::JobKey
java_import Java::MyscheduleQuartzExtraJob::LoggerJob
java_import Java::MyscheduleQuartzExtra::SchedulerTemplate

job = SchedulerTemplate.create_job_detail(JobKey.job_key("jobWithMutltipleTriggers"), LoggerJob.java_class, true, nil)
scheduler.add_job(job, false)
trigger1 = SchedulerTemplate.create_simple_trigger("trigger1", -1, 60 * 60 * 1000) # hourly trigger
trigger1.set_job_key(job.get_key())
scheduler.schedule_job(trigger1)
trigger2 = SchedulerTemplate.create_simple_trigger("trigger2", -1, 60 * 1000) # minutely trigger
trigger2.set_job_key(job.get_key())
scheduler.schedule_job(trigger2)

# Using Java org.scheduler.Scheduler API
# ===================================================
require 'java'
java_import Java::OrgQuartz::JobBuilder
java_import Java::OrgQuartz::TriggerBuilder
java_import Java::OrgQuartz::SimpleScheduleBuilder
java_import Java::MyscheduleQuartzExtraJob::LoggerJob

quartzScheduler = scheduler.getScheduler()
job = JobBuilder.
	newJob(LoggerJob.java_class).
	withIdentity("hourlyJob2").
	build()
trigger = TriggerBuilder.
	newTrigger().
	withSchedule(
		SimpleScheduleBuilder.repeatHourlyForever()).
	build()
quartzScheduler.scheduleJob(job, trigger)

File: myschedule-web-config/src/main/resources/myschedule/web/templates/scripts/simpleJobs.js
// JavaScript Examples

// Using myschedule.quartz.extra.SchedulerTemplate API
// ===================================================
importClass(Packages.myschedule.quartz.extra.job.LoggerJob);
importClass(Packages.myschedule.quartz.extra.job.ScriptingJob);
importClass(Packages.java.lang.System);
importClass(Packages.java.util.HashMap);

//Schedule hourly job
scheduler.scheduleSimpleJob("hourlyJob", -1, 60 * 60 * 1000, LoggerJob);

// Schedule minutely job
scheduler.scheduleSimpleJob("minutelyJob", -1, 60 * 1000, LoggerJob);

// Schedule secondly job
scheduler.scheduleSimpleJob("secondlyJob", -1, 1000, LoggerJob);

// Schedule secondly job that repeat total of 3 times.
scheduler.scheduleSimpleJob("secondlyJobRepeat3", 3, 1000, LoggerJob);

// Schedule onetime job that run immediately
scheduler.scheduleSimpleJob("onetimeJob", 1, 0, LoggerJob);

// Schedule hourly job with job data and start time of 20s delay.
dataMap = HashMap();
dataMap.put('ScriptEngineName', 'JavaScript');
dataMap.put('ScriptText', 
		'logger.info("I am a script job...");\n' +
		'Packages.java.lang.Thread.sleep(700);\n' +
		'logger.info("I am done.");\n');
scheduler.scheduleSimpleJob("hourlyJobWithStartTimeDelay", -1, 60 * 60 * 1000, ScriptingJob, 
		dataMap, 
		Packages.java.util.Date(System.currentTimeMillis() + 20 * 1000));

// Schedule one job with multiple triggers
importClass(Packages.myschedule.quartz.extra.job.LoggerJob);
importPackage(Packages.org.quartz);
job = scheduler.createJobDetail("jobWithMutltipleTriggers", "DEFAULT", LoggerJob, true, null);
scheduler.addJob(job, false);
trigger1 = scheduler.createSimpleTrigger("trigger1", -1, 60 * 60 * 1000); // hourly trigger
trigger1.setJobName(job.getName());
scheduler.scheduleJob(trigger1);
trigger2 = scheduler.createSimpleTrigger("trigger2", -1, 60 * 1000); // minutely trigger
trigger2.setJobName(job.getName());
scheduler.scheduleJob(trigger2);

// Using Java org.scheduler.Scheduler API
// ===================================================
importClass(Packages.myschedule.quartz.extra.job.LoggerJob);
importPackage(Packages.org.quartz);
quartzScheduler = scheduler.getScheduler();
job = JobDetail("hourlyJob2", "DEFAULT", LoggerJob);
trigger = SimpleTrigger("hourlyJob2", "DEFAULT", -1, 60 * 60 * 1000);
quartzScheduler.scheduleJob(job, trigger);
